Form Recognizer v3.1 GA (2023-07-31)

bojunehsu opened the pull request:
# Data Plane API - Pull Request
<!-- 🚨🚨🚨 Replace this line with a summary and reason for these changes to your data plane API 🚨🚨🚨 -->

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [X] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document: https://mixedrealitywiki.com/pages/viewpage.action?pageId=231645912
* Previous API Spec Doc: https://github.com/Azure/azure-rest-api-specs-pr/blob/main/specification/cognitiveservices/data-plane/FormRecognizer/preview/2023-02-28-preview/FormRecognizer.json
* Updated paths: None

### Details
* (Not GA breaking) Remove the queryFields query parameter.  It was first introduced in v3.1 preview.
* (Not GA breaking) Remove the keyValuePair.commonName property.  It was first introduced in v3.1 preview.
* (Not GA breaking, preview breaking) Office/HTML support was first introduced in v3.0 preview.3, but removed from v3.0 GA.  In v3.1 GA, we will introduce it in GA for the first time.  Relative to previous previews, we are removing support for embedded images (page.{kind,images}).  We also change the behavior to return empty arrays for unsupported features instead of an error.
* (Not GA breaking, preview breaking) Renamed enum values of the features query parameter to better reflect feature behavior and map to the corresponding property name in the response JSON.  For example, renamed "ocr.formula" to "formulas" to better reflect that we may also extract formulas from docx/html in the future.  The new name better indicates that the results are expected to be found under $.pages[*].formulas.
* (Not GA breaking) Add more enum values to the features query parameter to allow customers to enable optional features across more models (ex. barcodes, languages, keyValuePairs).
* Refined documentation.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

bojunehsu committed:
Copy 2023-02-28-preview to 2023-07-31

bojunehsu committed:
v3.1 GA changes

mikekistler reviewed (approved):
Looks good! 👍 

Thanks for adding all the x-ms-enum descriptions!

johanste commented on specification/cognitiveservices/data-plane/FormRecognizer/stable/2023-07-31/FormRecognizer.json:
```diff
@@ -0,0 +1,3336 @@
+        }
+      }
+    },
+    "ClassifierDocumentTypeDetails": {
```
This should be a discriminated polymorphic type.

bojunehsu committed:
Update documentation for REST docs

bojunehsu commented:
@jhendrixMSFT

I believe the reported breaking changes are all false positives and reflect changes already approved in the latest preview. As compared with [latest preview](https://github.com/Azure/azure-rest-api-specs-pr/pull/12990/checks?check_run_id=14571458557), we have the following intentional changes:

* Renamed _AzureBlobFileListSource_ to _AzureBlobFileListContentSource_ (introduced in latest preview).
* _queryFields_ and _commonName_ are both intentionally removed, as they are not ready for GA (introduced in latest preview).
* Renamed and added a few enum values for _fields_ query parameter (introduced in latest preview).
* _DocumentPage.{kind,images}_ are intentionally removed (introduced in 2022-06-30-preview, but never GAed).

There is insufficient information in the error messages for me to figure out why Swagger APIView is failing. Will appreciate any insights you may have.
